src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/FaultDetectionMetricsCollector.java

```diff
@@ -0,0 +1,153 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsConfiguration;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.MetricsProcessor;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics;
+import com.google.gson.Gson;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.tools.StringUtils;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.stream.Collectors;
+
+import static com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.PerformanceAnalyzerMetrics.addMetricEntry;
+
+public class FaultDetectionMetricsCollector extends PerformanceAnalyzerMetricsCollector implements MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL = MetricsConfiguration.CONFIG_MAP.
+            get(FaultDetectionMetricsCollector.class).samplingInterval;
+    private static final int KEYS_PATH_LENGTH = 3;
+    private static final Logger LOG = LogManager.getLogger(FaultDetectionMetricsCollector.class);
+    private static final String FAULT_DETECTION_HANDLER_NAME =
+            "com.amazon.opendistro.elasticsearch.performanceanalyzer.collectors.ClusterFaultDetectionStatsHandler";
+    private static final String FAULT_DETECTION_HANDLER_METRIC_QUEUE = "metricQueue";
+    private StringBuilder value;
+
+    public FaultDetectionMetricsCollector() {
+        super(SAMPLING_TIME_INTERVAL, "FaultDetectionMetrics");
+        value = new StringBuilder();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    void collectMetrics(long startTime) {
+        Class<?> faultDetectionHandler = null;
+        try {
+            faultDetectionHandler = Class.forName(FAULT_DETECTION_HANDLER_NAME);
+        } catch (ClassNotFoundException e) {
+            LOG.debug("No Handler Detected for Fault Detection. Skipping FaultDetectionMetricsCollector");
+        }
```

Comment:
forName is registered to throw only ClassNotFoundException. What other exception is expected to be thrown here ?

Reply:
Interesting. As per the JavaDoc it can throw LinkageError as well as the ExceptionInInitializerError. ExceptionInInitializerError is a subclass for LinkageError and both of them are thrown when there are compatibility issues after compilation.